### ISSUE TITLE
CI: add a job to create a gdal-docs Docker image

### ISSUE THREAD

**Issue by rouault** [OSGeo/gdal#7880]
.github/workflows/doc_build.yml depends on "container: ghcr.io/osgeo/proj-docs" which is currently built within the OSGeo/PROJ repository, but since the RTD migration of PROJ, PROJ doesn't use it anymore
Ultimately we should move that content from OSGeo/PROJ to here
Cf https://github.com/OSGeo/PROJ/pull/3743#issuecomment-1567140901

**Comment by hobu** [comment]
Is there enthusiasm for migrating GDAL to RTD too, or should we keep things as-is? The biggest win of RTD is convenient PR builds, but I don't have a strong opinion about whether or not GDAL should migrate.

**Comment by rouault** [comment]
There could be interest in migrating to RTD. People are regularly confused to see mentions of GDAL versions which aren't released yet as we publish the master doc.

**Comment by mwtoews** [comment]
I'd be keen to see a similar migration happen, mostly for the benefit of multi-version docs. I could possibly help out in a similar capacity to PROJ's migration, although I don't have a long perspective on any challenges building docs for older releases of GDAL.

It would be wise to canvas this idea on the mail list to see is there is support before going much further.

**Comment by rouault** [comment]
> although I don't have a long perspective on any challenges building docs for older releases of GDAL.

I don't think we need to exhaust ourselves building docs for older releases. Probably starting with just GDAL 3.6 and 3.7 should be fine IMHO
